### DISCUSSION TITLE
GetNumConformers() returns 0 for all molecules?

### DISCUSSION THREAD

**Discussion by fyy623** [rdkit/rdkit#7040] [FAQ]
This function seems to be not working for any kinds of molecules. 

```print(f"Number of conformers: {Chem.MolFromSmiles('COC1=CC(=NC=C1)C2=NC=CC(=C2)OC').GetNumConformers()}")```

Below is the molecule I am using, which obviously has more than 1 conformers.

https://pubchem.ncbi.nlm.nih.gov/compound/2733927#section=3D-Conformer

![image](https://github.com/rdkit/rdkit/assets/59494795/041f954c-4c79-40af-aee5-30c689b29a9b)

**Comment by dehaenw** [comment]
You have no conformers because you didn't calculate any, nor provide 3D coordinates (the molecule was instantiated from SMILES only). This function does not return the total number of conformers (which is anyway subject to how you define cutoffs), it returns the amount of conformations you have stored in your molecule object. If you'd like to generate conformations in RDKit here is more info: https://www.rdkit.org/docs/GettingStartedInPython.html#working-with-3d-molecules

**Reply by fyy623** [reply]
I see, I am new to rdkit and thought a mol object contains all basic information a molecule has already...

Just a similar follow-up: is it same that to get MolToPDBFile work, I need to first embed a conformer into the mol object using AllChem.EmbedMolecule()?

**Reply by dehaenw** [reply]
yes, you'll also have to embed this if you dont want the coordinates to all be (0,0,0). beware that PDBfiles are considered to be a suboptimal format for small molecules (often having various problems with bond orders, connectivity, hydrogen count, etc depending on the source of the PDB and the program you are using for reading it)

**Reply by fyy623** [reply]
I see, what would be the optimal format for small molecules, say less than 50 atoms?

**Reply by dehaenw** [reply]
i'd recommend to use sdf.

**Reply by bp-kelley** [reply]
Here is how you write out a sdf file
```
writer = Chem.SDWriter("my.sdf")
writer.write(mol)
writer.close()
```

And for reading:
```
for mol in Chem.SDMolSupplier("my.sdf"):
     ...
```
